src/rdkafka.h

```diff
@@ -9848,6 +9869,221 @@ RD_EXPORT void rd_kafka_DeleteAcls(rd_kafka_t *rk,
+ *    Controls how long \c rdkafka will wait for the request to complete.
+ *
+ * @remark The result event type emitted on the supplied queue is of type
+ *         \c RD_KAFKA_EVENT_ELECTLEADERS_RESULT
+ */
+RD_EXPORT void rd_kafka_ElectLeaders(rd_kafka_t *rk,
+                                     rd_kafka_ElectLeaders_t *elect_leaders,
+                                     const rd_kafka_AdminOptions_t *options,
+                                     rd_kafka_queue_t *rkqu);
+
+/**
+ * @brief A struct consisting of Topic Partition result
+ *        including custom error message.
+ */
+typedef struct rd_kafka_topic_partition_result_s
+    rd_kafka_topic_partition_result_t;
+
+/**
+ * @brief  A struct representing result of elect leaders admin operation.
+ */
+typedef struct rd_kafka_ElectLeadersResult_s rd_kafka_ElectLeadersResult_t;
+
+/**
+ * @brief Get the topic name from the rd_kafka_topic_partition_result_t object.
+ *
+ * @param partition_result The rd_kafka_topic_partition_result_t object.
+ *
+ * @returns the topic name from the rd_kafka_topic_partition_result_t object.
+ */
+RD_EXPORT const char *rd_kafka_topic_partition_result_topic(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @brief Get the partition number from the rd_kafka_topic_partition_result_t
+ * object.
+ *
+ * @param partition_result The rd_kafka_topic_partition_result_t object.
+ *
+ * @returns the partition number from the rd_kafka_topic_partition_result_t
+ * object.
+ */
+RD_EXPORT int32_t rd_kafka_topic_partition_result_partition(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @brief Get the error code from the rd_kafka_topic_partition_result_t object.
+ *
+ * @param partition_result The rd_kafka_topic_partition_result_t object.
+ *
+ * @returns the error code from the rd_kafka_topic_partition_result_t object.
+ */
+RD_EXPORT rd_kafka_resp_err_t rd_kafka_topic_partition_result_error(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @brief Get the error string from the rd_kafka_topic_partition_result_t
+ * object.
+ *
+ * @param partition_result The rd_kafka_topic_partition_result_t object.
+ *
+ * @returns the error string or NULL if there was no error.
+ */
+RD_EXPORT const char *rd_kafka_topic_partition_result_error_string(
+    const rd_kafka_topic_partition_result_t *partition_result);
+
+/**
+ * @brief Get the rd_kafka_topic_partition_result_t object at index \p idx from
+ * the rd_kafka_topic_partition_result_t array.
+ *
+ * @param result The array of rd_kafka_topic_partition_result_t objects.
+ * @param idx The index of the object to be retrieved.
+ *
+ * @returns the rd_kafka_topic_partition_result_t object at index \p idx from
+ * the rd_kafka_topic_partition_result_t array.
+ *
+ * @remark The index \p idx must be less than the size of the array.
+ *
+ *
+ */
+RD_EXPORT const rd_kafka_topic_partition_result_t *
+rd_kafka_topic_partition_result_by_idx(
```

Comment:
I think this is specific to GoLang and must be kept in GoLang only in the comments like other similar functions